cfme/common/provider.py

```diff
@@ -35,15 +89,411 @@ def get_yaml_data(self):
+        tb.select("Configuration", "Refresh Relationships and Power States", invokes_alert=True)
+        sel.handle_alert(cancel=False)
+
+    def _load_details(self):
+        if not self._on_detail_page():
+            sel.force_navigate('{}_provider'.format(self.page_name), context={'provider': self})
+
+    def get_detail(self, *ident):
+        """ Gets details from the details infoblock
+
+        The function first ensures that we are on the detail page for the specific provider.
+
+        Args:
+            *ident: An InfoBlock title, followed by the Key name, e.g. "Relationships", "Images"
+        Returns: A string representing the contents of the InfoBlock's value.
+        """
+        self._load_details()
+        return details_page.infoblock.text(*ident)
+
+    def load_all_provider_instances(self):
+        self.load_all_provider_vms()
+
+    def load_all_provider_vms(self):
+        """ Loads the list of instances that are running under the provider.
+
+        If it could click through the link in infoblock, returns ``True``. If it sees that the
+        number of instances is 0, it returns ``False``.
+        """
+        sel.force_navigate('{}_provider'.format(self.page_name), context={'provider': self})
+        if details_page.infoblock.text("Relationships", self.vm_name) == "0":
+            return False
+        else:
+            sel.click(details_page.infoblock.element("Relationships", self.vm_name))
+            return True
+
+    def load_all_provider_images(self):
+        self.load_all_provider_templates()
+
+    def load_all_provider_templates(self):
+        """ Loads the list of images that are available under the provider.
+
+        If it could click through the link in infoblock, returns ``True``. If it sees that the
+        number of images is 0, it returns ``False``.
+        """
+        sel.force_navigate('{}_provider'.format(self.page_name), context={'provider': self})
+        if details_page.infoblock.text("Relationships", self.template_name) == "0":
+            return False
+        else:
+            sel.click(details_page.infoblock.element("Relationships", self.template_name))
+            return True
+
+    """
+    From cloud
+    def _on_detail_page(self):
+        "" Returns ``True`` if on the providers detail page, ``False`` if not.""
+        return sel.is_displayed('//div[@class="dhtmlxInfoBarLabel-2"][contains(., "%s")]'
+                                % self.name)
+    """
+
+    def _on_detail_page(self):
+        """ Returns ``True`` if on the providers detail page, ``False`` if not."""
+        ensure_browser_open()
+        return sel.is_displayed(
+            '//div[@class="dhtmlxInfoBarLabel-2"][contains(., "%s (Summary)")]' % self.name)
+
+    # @classmethod
+    # def get_from_config(provider_key):
+    #    return utils.provider.get_crud(provider_key)
+
+
+@fill.method((Form, BaseProvider.Credential))
```

Comment:
Now we have no SCVMM special fill method as the Credential object is no different from the others. It can be handled in the same way and just inspected for the extra domain attribute.